### PR TITLE
Align Datastructures to Spec and Add Fixed-Length Tuple Support

### PR DESCRIPTION
**Pull request by schroedingerscode**
## PR Description
This PR rewrites the datastructures to conform to the v0.5.1 Beacon Chain spec, and also adds support for "tuples" (fixed-length lists), which are serialized without the length mixin.

### CODE REVIEW

**Comment by schroedingerscode** [comment]
Quick first pass to help rebasing on top of the epoch upgrade. More to come when Cava master build hash is available.

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
 public class Fork {
 
+  // TODO previous_version should be a bytes4 (this has serialization impacts)
   private Bytes previous_version;
+  // TODO previous_version should be a bytes4 (this has serialization impacts)
```
Comment (cemozerr) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java]: should this be current_version?

File: gradle/versions.gradle [hunk 2]
```diff
@@ -45,7 +45,7 @@ dependencyManagement {
       entry 'cava-junit'
       entry 'cava-kv'
       entry 'cava-rlpx'
-      entry 'cava-ssz'
+      // entry 'cava-ssz'
```
Comment (akhila-raju) [inline gradle/versions.gradle:49]: Would you mind adding a comment explaining why this is commented out?

**Review by akhila-raju — CHANGES_REQUESTED** [review]
Please add a comment explaining why cava is commented out in the versions.gradle file

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java (hunk 1 above)
Comment (dangerousfood) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java]: I think this implementation is actually correct https://github.com/ethereum/eth2.0-specs/blob/v0.5.1/specs/core/0_beacon-chain.md#fork

Comment (akhila-raju) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java]: I think @cemozerr is referring to the name on the comment @dangerousfood, it should be current_version

File: gradle/versions.gradle (hunk 2 above)
Comment (dangerousfood) [reply gradle/versions.gradle:49]: 19b9e56

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java (hunk 1 above)
Comment (dangerousfood) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/Fork.java]: 84784e1

**Review by akhila-raju — APPROVED** [review]
LGTM

**Pull request merged**